### DISCUSSION TITLE
[dialog] unable to reference dialog node

### DISCUSSION THREAD

**Discussion by LarsDenBakker** [ing-bank/lion#1266] [General]
The dialog element has no public API to get a reference to the content node. There is a `_overlayContentNode` property but it is private.

**Comment by tlouisse** [comment]
All 'node/element getters' are private (or actually protected to be really precise), so they are only accessible for Subclassers.

For some of them it might make sense to make them public (we could add an extra getter that calls the extisting one) but I would like to keep this limited. 
Exposing all of them would be too much(there are a lot in our components in general, so it would be an 'api explosion') and it would make us very inflexible with refactorings etc.

@LarsDenBakker is there a reason you need it for this particular getter? (It's the element you provide via content projection I think, right?)

**Comment by jorenbroekema** [comment]
I think this is a good enhancement but I agree with Thijs that we have to factor in pros and cons, the cons being exposing this API makes us less flexible.

For us to pick this up we definitely will need to hear about use cases where you really need to grab the _overlayContentNode as a public property. Will move this to discussions
